firedrake/preconditioners/hypre_ams.py

```diff
@@ -44,14 +43,8 @@ def initialize(self, obj):
         if zero_beta:
             pc.setHYPRESetBetaPoissonMatrix(None)
 
-        # Build constants basis for the Nedelec space
-        cvecs = []
-        for i in range(mesh.cell_dimension()):
-            direction = [1.0 if i == j else 0.0 for j in range(mesh.cell_dimension())]
-            c = project(Constant(direction), V)
-            with c.vector().dat.vec_ro as cvec:
-                cvecs.append(cvec)
-        pc.setHYPRESetEdgeConstantVectors(*cvecs)
```

Comment:
I guess you did this because the convergence is better? Is that right?

Reply:
For some reason, project leads to a `DIVERGED_PCSETUP_FAILED` error for a fine mesh. It works with an older version of firedrake and still works for a coarse mesh. When I originally implemented the code, interpolation on edge elements was not implemented, if i remember correctly. Furthermore, using the coordinates led to bad convergence. I think the problem was that the dofs were not ordered correctly. 
Now that setCoordinates yields good convergence, I think this should be the preferred version, at least for the lowest-order case.

Reply:
OK thanks. I suspect the PCSETUP_FAILED divergence is because we switched to using a direct solver by default (even for projections) and so for very big problems that might cause OOM or other issues (see #1900).